src/ehttpc.erl

```diff
@@ -157,6 +170,21 @@ init([Pool, Id, Opts]) ->
     true = gproc_pool:connect_worker(ehttpc:name(Pool), {Pool, Id}),
     {ok, State}.
 
+handle_call({health_check, _}, _From, State = #state{gun_state = up}) ->
+    {reply, ok, State};
+handle_call({health_check, Timeout}, _From, State = #state{gun_state = down}) ->
+    case open(State) of
```

Comment:
should not open if the process is currently waiting for `gun_up`, simply enter the wait loop.